### PR TITLE
Update parsing for new click-in-text API

### PR DESCRIPTION
**Pull request by Madoshakalaka**
A bug mentioned in the meeting is fixed: now lemma definitions are correctly shown
New actions runs buttery smooth. All integration tests pass for this PR and linter, release didn't run. (We'll see if they run as expected when this gets merged)

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1 +1 @@
-{"video": false}
+{"video": false, "defaultCommandTimeout": 10000 }
```
Comment (Madoshakalaka) [inline cypress.json:1]: requests has been slow for me maybe cuz I'm testing from China. It doesn't usually takes this long though

File: cypress/integration/copy_translation_to_clipboard.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 // copy button doesn't work
-describe('Clipboard', function() {
-  it.skip('copies the last translation to clipboard', function() {
+describe.skip('Clipboard', function() {
+  it('copies the last translation to clipboard', function() {
```
Comment (Madoshakalaka) [inline cypress/integration/copy_translation_to_clipboard.js:3]: A previous action run shows we have this test pending forever...and it failed the test (non zero return)!?? Although this test file contains only this skipped test. This part of code looks dubious to me.

we also ran Cypress like `cypress run --spec cypress/integrations/*.js`. And this usage is warned against in this previous run. Cypress complains that this invocation also works but it's like recommended against blah blah.

In this PR, it.skip is lifted to describe.skip.
And cypress invocation has been corrected to `cypress run`
I don't know which one helped but the tests pass!

File: cypress/integration/oops_option.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 describe('Always show oops setting', function() {
   it('shows oops message', function() {
     cy.visit('localhost:8080/test.html')
-    cy.get('#non_cree_hello').click()
+    cy.get('#Chinese').click()
```
Comment (Madoshakalaka) [inline cypress/integration/oops_option.js:5]: We used to produce "Oops...translation not found" for _hello_
Now click-in-text translates English too.

File: test.html [hunk 4]
```diff
@@ -95,8 +95,9 @@
   <body>
 
 
-  <span id="test1">niskak</span> <br>
-  <span id="non_cree_hello">hello</span>
+  <span id="test1">wâpamêw</span> <br>
+  <span id="English">hello</span> <br>
+  <span id="Chinese">你好</span> <br>
```
Comment (Madoshakalaka) [inline test.html:100]: 你好 ( ni hao ) means hello in Chinese

File: test.html [hunk 5]
```diff
@@ -106,7 +107,7 @@
   <label for="test_input">test input:</label><input id="test_input">
   <hr>
   <button class="big-ass-button" id="tat_button">Open TAT</button>
-  <button class="big-ass-button" id="copy_button">Copy shit</button>
+  <button class="big-ass-button" id="copy_button">Copy things</button>
```
Comment (Madoshakalaka) [inline test.html:110]: I felt uncomfortable about this...pls don't git blame though

File: lib/transover_core.js [hunk 6]
```diff
@@ -7,22 +7,21 @@ import XRegExp from 'xregexp/src'
 const Core = {}
```
Comment (Madoshakalaka) [inline lib/transover_core.js:7]: most of the changes in this file is due to formatting. I'll annotate the meaning changes below

File: lib/transover_core.js [hunk 7]
```diff
@@ -36,12 +35,12 @@ function reloadAndApplyOptionsOnTabSwitch (loadAndApplyOptions) {
 Core.callAPI = (word, onresponse, sendResponse) => {
 
   const options = {
-    url: 'http://sapir.artsrn.ualberta.ca/cree-dictionary/_translate-cree/'+word,
+    url: 'https://sapir.artsrn.ualberta.ca/cree-dictionary/click-in-text?q=' + word,
```
Comment (Madoshakalaka) [inline lib/transover_core.js:38]: now query params are used instead

File: cypress/integration/copy_translation_to_clipboard.js (hunk 2 above)
Comment (eddieantonio) [reply cypress/integration/copy_translation_to_clipboard.js:3]: Haha! Yeah, I'm not sure why `cypress run --spec` was being run that way. Honestly, you could probably just delete this test for now.

File: lib/transover_core.js [hunk 8]
```diff
@@ -51,50 +50,62 @@ Core.callAPI = (word, onresponse, sendResponse) => {
...
+    data.results.forEach(function (result) {
+
+      const definition_list = []
+
+      result.lemma_wordform.definitions.forEach(function (definition) {
+        definition_list.push(definition.text + '; ' + definition.source_ids.join(' '))
+      }
+      )
+      output.push({pos: result.lemma_wordform.text + result.lemma_wordform.analysis, meanings: definition_list})
+    })
```
Comment (Madoshakalaka) [inline lib/transover_core.js:87]: This is the API parsing part. I'll convert the todo to a issue

File: cypress/integration/translate_by_clicking.js [hunk 9]
```diff
@@ -1,12 +1,12 @@
 describe('Default settings', function() {
-  it('shows popup for niskak', function() {
+  it('shows popup for cree', function() {
     cy.visit('localhost:8080/test.html')
     // cy.get('#test1').move()
```
Comment (eddieantonio) [inline cypress/integration/translate_by_clicking.js:4]: I wonder what this was doing here...

```suggestion
```

File: test.html (hunk 5 above)
Comment (eddieantonio) [reply test.html:110]: pfhahhahhahah! I love the class name,

File: lib/transover_core.js (hunk 6 above)
Comment (eddieantonio) [reply lib/transover_core.js:7]: 💀  I was wondering why this was such a large diff :/

File: lib/transover_core.js (hunk 7 above)
Comment (eddieantonio) [inline lib/transover_core.js:38]: I'm not 100% sure, but you _might_ need to encode the `word` using [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)

```suggestion
    url: 'https://sapir.artsrn.ualberta.ca/cree-dictionary/click-in-text?q=' + encodeURIComponent(word),
```

File: lib/transover_core.js [hunk 10]
```diff
@@ -51,50 +50,62 @@ Core.callAPI = (word, onresponse, sendResponse) => {
 
 
 Core.parseAPIResponse = (data, word, sendResponse) => {
+  // for click-in-text api response, see this schema file:
+  // https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/blob/master/CreeDictionary/API/schema.py
```
Comment (eddieantonio) [inline lib/transover_core.js:54]: ohhhh, very nice!

**Review by eddieantonio — APPROVED** [review]
LGTM! I'm not certain if something calls `encodeURIComponent` for you, but generally, you'll want to call that if you're constructing a URL with query params 👍🏼

File: lib/transover_core.js (hunk 7 above)
Comment (eddieantonio) [reply lib/transover_core.js:38]: Actually, the more I look at this, the more I'm convinced you _do_ need `encodeURIComponent()` here...

**Pull request merged**